src/codeeditor/index.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  Token
+} from 'phosphor/lib/core/token';
+
+import {
+  CodeEditor
+} from './editor';
+
+export * from './widget';
+export * from './editor';
+
+/* tslint:disable */
+/**
+ * The editor factory token.
+ */
+export
+const IEditorFactory = new Token<IEditorFactory>('jupyter.services.editorfactory');
+/* tslint:enable */
+
+/**
+ * The editor factory interface.
+ */
+export
+interface IEditorFactory {
+
+  /**
+   * Create a new editor for inline code.
+   */
+  newInlineEditor(host: HTMLElement, options: CodeEditor.IOptions): CodeEditor.IEditor;
```

Comment:
Do we really need it? Is not an instance can be provided by a editor specific renderer of a plugin? E.g. for notebook, CodeMirrorRenderer can create CodeMirrorEditor.